lib/streamlit/runtime/metrics_util.py

```diff
@@ -15,10 +15,50 @@
 import os
 import threading
 import uuid
+import contextlib
+import enum
+import sys
+import inspect
+from collections.abc import Sized
+from functools import wraps
+from timeit import default_timer as timer
+from typing import Any, Callable, List, Optional, TypeVar, cast, Set
 
-from typing import Optional
+from typing_extensions import Final
 
 from streamlit import util
+from streamlit import config
+from streamlit.logger import get_logger
+from streamlit.proto.PageProfile_pb2 import Argument, Command
+from streamlit.proto.ForwardMsg_pb2 import ForwardMsg
+from streamlit.runtime.scriptrunner import get_script_run_ctx
+
+LOGGER = get_logger(__name__)
+
+# Limit the number of commands to keep the page profile message small
+# since Segment allows only a maximum of 32kb per event.
+_MAX_TRACKED_COMMANDS: Final = 200
+# A mapping to convert from the actual name to preferred/shorter representations
+_OBJECT_NAME_MAPPING: Final = {
+    "streamlit.delta_generator.DeltaGenerator": "DG",
+    "pandas.core.frame.DataFrame": "DataFrame",
```

Comment:
seems inconsistent to have camelCase in the string value since it's a python mapping. Not sure if it REALLY matters but just a thought.

Reply:
In that case it refers to object types, which are written with camelCase also in python: e.g. `DataFrame`.